app/src/modules/network/network.c

```diff
@@ -123,6 +123,85 @@ static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
 	}
 }
 
+static void lte_lc_evt_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS: {
```

Comment:
We only need this case, the other ones can be ignored. Future improvement would be to add this check to the nrf_modem_net_if implementation.